packages/ember-testing/tests/helpers_test.js

```diff
@@ -96,1168 +103,1186 @@ class HelpersApplicationTestCase extends HelpersTestCase {
+        this.application.setupForTesting();
+      });
 
-    assert.equal(
-      this.application.testing, true,
-      'Application instance is set to testing.'
-    );
-  }
+      let routerInstance = this.applicationInstance.lookup('router:main');
+      assert.equal(routerInstance.location, 'none');
+    }
 
-  [`@test Ember.Application.setupForTesting leaves the system in a deferred state.`](assert) {
-    this.runTask(() => {
-      this.createApplication();
-      this.application.setupForTesting();
-    });
+    [`@test Ember.Application.setupForTesting sets the application to 'testing'`](assert) {
+      this.runTask(() => {
+        this.createApplication();
+        this.application.setupForTesting();
+      });
 
-    assert.equal(
-      this.application._readinessDeferrals, 1,
-      'App is in deferred state after setupForTesting.'
-    );
-  }
+      assert.equal(
+        this.application.testing, true,
+        'Application instance is set to testing.'
+      );
+    }
 
-  [`@test App.reset() after Application.setupForTesting leaves the system in a deferred state.`](assert) {
-    this.runTask(() => {
-      this.createApplication();
-      this.application.setupForTesting();
-    });
+    [`@test Ember.Application.setupForTesting leaves the system in a deferred state.`](assert) {
+      this.runTask(() => {
+        this.createApplication();
+        this.application.setupForTesting();
+      });
 
-    assert.equal(
-      this.application._readinessDeferrals, 1,
-      'App is in deferred state after setupForTesting.'
-    );
+      assert.equal(
+        this.application._readinessDeferrals, 1,
+        'App is in deferred state after setupForTesting.'
+      );
+    }
 
-    this.application.reset();
+    [`@test App.reset() after Application.setupForTesting leaves the system in a deferred state.`](assert) {
+      this.runTask(() => {
+        this.createApplication();
+        this.application.setupForTesting();
+      });
 
-    assert.equal(
-      this.application._readinessDeferrals, 1,
-      'App is in deferred state after setupForTesting.'
-    );
-  }
+      assert.equal(
+        this.application._readinessDeferrals, 1,
+        'App is in deferred state after setupForTesting.'
+      );
 
-  [`@test #setupForTesting attaches ajax listeners`](assert) {
-    let documentEvents = jQuery._data(document, 'events') || {};
+      this.application.reset();
 
-    assert.ok(
-      documentEvents['ajaxSend'] === undefined,
-      'there are no ajaxSend listers setup prior to calling injectTestHelpers'
-    );
-    assert.ok(
-      documentEvents['ajaxComplete'] === undefined,
-      'there are no ajaxComplete listers setup prior to calling injectTestHelpers'
-    );
+      assert.equal(
+        this.application._readinessDeferrals, 1,
+        'App is in deferred state after setupForTesting.'
+      );
+    }
 
-    setupForTesting();
+    // [`@test #setupForTesting attaches ajax listeners`](assert) {
```

Comment:
If we want to bring back these tests, we would have to add back the `jquery.on` listeners in `setup_for_testing`.  I don't think there is a way to track all of the events listening in a cross browser way (`getEventListeners`; however, I think no IE).   We could add an explicit trackEventLIsteners method but I don't think we would want to do that either.
 
However, it seems to me that the below tests cover the functionality in `setup_for_testing` and these commented out tests could be deleted.

https://github.com/emberjs/ember.js/blob/master/packages/ember-testing/tests/helpers_test.js#L1074

Reply:
@rwjblue lmk what you think about this remaining issue.

Reply:
@snewcomer - Can you delete them?